src/Appear.js

```diff
@@ -24,51 +18,46 @@ export default class Appear extends React.Component {
 
   handleKeyDown = e => {
     if (e.metaKey || e.ctrlKey || e.shiftKey || e.altKey) return
+    if (!this.props.slide.active) return
+    const { children } = this.props
+    const { update } = this.props.slide
     switch (e.key) {
       case 'ArrowDown':
         e.preventDefault()
-        this.setState(state => {
-          return {
-            fragmentStep:
-              state.fragmentStep < state.fragments.length - 1
-                ? state.fragmentStep + 1
-                : state.fragmentStep
-          }
-        })
+        update(state => ({
```

Comment:
I think there's two different approaches to this state updating pattern that I don't mind, either:
1. Passing a generic `update` method down and importing "updater" functions from a common module, e.g. `update(decrementStep)` (not sure about the verb)
2. Passing more explicitly names methods down, e.g. `this.props.deck.decrementStep`

Reply:
Sure, what would be the preferred way? Maybe passing down `update` might have an advantage of enabling custom components to be written. For now, we can also go for the second approach and think about a more generic approach when we really needed it.
Either way it's definitely a good idea to move this functions into a common module.

Reply:
Yeah, I'm okay with trying 1 to start with – it gives you a little more flexibility